src/vs/workbench/contrib/chat/browser/chatWidget.ts

```diff
@@ -534,10 +534,11 @@ export class ChatWidget extends Disposable implements IChatWidget {
 		this._dynamicMessageLayoutData = { numOfMessages: numOfChatTreeItems, maxHeight };
 		this._register(this.renderer.onDidChangeItemHeight(() => this.layoutDynamicChatTreeItemMode()));
 
-		const mutableDisposable = this._register(new MutableDisposable());
+		let cts: CancellationTokenSource | undefined;
 		this._register(this.tree.onDidScroll((e) => {
-			mutableDisposable.value = dom.scheduleAtNextAnimationFrame(() => {
-				mutableDisposable.dispose();
+			cts?.cancel();
+			cts = new CancellationTokenSource();
+			dom.scheduleAtNextAnimationFrame(() => {
 				if (!e.scrollTopChanged || e.heightChanged || e.scrollHeightChanged) {
 					return;
 				}
```

Comment:
I think this might be buggy.
Imagine onDidScroll fires twice before `scheduleAtNextAnimationFrame` kicks in.
The first time wouldn't take the return here, but the second time does.
However, the second time cancels the scheduled animation frame from the first time, thus `layout` wouldn't be called at all.

It looks like you want to batch-process multiple layout calls until the next animation frame. For that, you need to combine the received deltas.